### ISSUE TITLE
Create Composite Views

### ISSUE THREAD

**Issue by Harbs** [apache/royale-asjs#257]
There are a number of components which are composites of other components. It's currently harder than it should be to change the appearance of parts of these components. We need "composite views" which allow declaration of the different "skin parts" to make it easier to change the appearance without rewriting a whole new view.

- [ ] ComboBox
- [ ] DateChooser
- [ ] Spinner
- [ ] TreeGrid
- [ ] DataGrid

**Comment by carlosrovira** [comment]
Also, think that maybe we should consider if it has sense to differentiate between normal and composite components. If we can change the view and skin parts easily in all components (even more basics like Button or TextInput), we'll not need different sets, or at least a component will extend the basic one to add some need ( a property for example,  "primary" in a button )

**Comment by Harbs** [comment]
Good point. It's worth figuring out if there's a PAYG way of making the views more swappable on simple components.

**Comment by aharui** [comment]
Basic Button was written to not have a replaceable view in JS because it is the smallest/fastest implementation.  If you want to make a "skinnable" component set, please do so.  IMO, Jewel is already that.  I think all of its views are replaceable.

**Comment by carlosrovira** [comment]
Actually Jewel Button is doing the same in that point as Basic Button. Both rely on "createElement". I'm working now in make IE11 support for Jewel. In next steps I want to have a look at SWF version for the current components (as we talked I'll be happy with components layout right as bounding boxes, throw events and work as expected. I can figure in that phase about skinning for the components that are still not using views and rely only in createElement.
